tests/vector_stores/test_elasticsearch.py

```diff
@@ -127,97 +127,143 @@ def test_instance_creation(index_name: str, elasticsearch_connection: Dict) -> N
 @pytest.mark.skipif(
     elasticsearch_not_available, reason="elasticsearch is not available"
 )
-def test_add_to_es_and_query(
+@pytest.mark.asyncio
+@pytest.mark.parametrize("use_async", [True, False])
+async def test_add_to_es_and_query(
     index_name: str,
     elasticsearch_connection: Dict,
     node_embeddings: List[TextNode],
+    use_async: bool,
 ) -> None:
     es_store = ElasticsearchStore(
         **elasticsearch_connection,
         index_name=index_name,
         distance_strategy="COSINE",
     )
-    es_store.add(node_embeddings)
-    res = es_store.query(
-        VectorStoreQuery(query_embedding=[1.0, 0.0, 0.0], similarity_top_k=1)
-    )
+    if use_async:
+        await es_store.async_add(node_embeddings)
+        res = await es_store.aquery(
+            VectorStoreQuery(query_embedding=[1.0, 0.0, 0.0], similarity_top_k=1)
+        )
+        await es_store.async_client.close()
```

Comment:
curious: why do we need to explicitly call close on the async client?

Reply:
so, it looks like we don't actually need it. In the previous version, where I had `AsyncElasticsearchStore` the tests would pass but then there'd be some warning about the client session not being closed. After explicitly calling `client.close()` back then would make the warning go away. 

I just tried removing all these statements from the tests; and, all test passed without such warnings.